src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -259,8 +259,13 @@ def require_package
     if params[:package].present?
       begin
         @package = Package.get_by_project_and_name(@project.to_param, params[:package],
-                                                   use_source: false, follow_project_links: true, follow_multibuild: true)
-      rescue APIException # why it's not found is of no concern :)
+                                                   follow_project_links: true, follow_multibuild: true)
+      rescue APIException => exception
+        flash[:error] = if exception.is_a?(Package::ReadSourceAccessError)
+          "You don't have access to the package \"#{params[:package]}\""
```

Comment:
You don't have access to the **sources** of this package.

Reply:
@hennevogel, changed